frontend/src/containers/Routing/index.tsx

```diff
@@ -1,17 +1,69 @@
-import React, { FunctionComponent } from 'react';
-import { Switch } from 'react-router-dom';
+import React, { useEffect } from 'react';
+import { connect } from 'react-redux';
+import { getAccessToken } from '../../common/helpers/storageHelper';
+import { Switch, Redirect } from 'react-router-dom';
 import { Routes } from '../../common/enums/Routes';
+import { IAppState } from '../../common/models/store/IAppState';
+import { fetchUserRoutine } from '../../scenes/Auth/routines';
 import PrivateRoute from '../PrivateRoute';
 import PublicRoute from '../PublicRoute';
 import Resources from '../../scenes/Resources/index';
 import Apps from '../../scenes/Apps/index';
+import Auth from '../../scenes/Auth/containers';
+import { IBindingAction } from '../../common/models/callback/IBindingAction';
 
-const Routing: FunctionComponent = () => (
-  <Switch>
-    <PrivateRoute path={Routes.Apps} component={Apps} />
-    <PrivateRoute path={Routes.Resources} component={Resources} />
-    <PublicRoute path={Routes.SignIn} component={() => <div>sign in</div>} />
-  </Switch>
-);
+interface IProps {
+    isLoading: boolean;
+    isAuthorized: boolean;
+    fetchUser: IBindingAction;
+}
 
-export default Routing;
+const Routing: React.FC<IProps> = ({
+  isLoading,
+  isAuthorized,
+  fetchUser
+}) => {
+  const hasToken = Boolean(getAccessToken());
+
+  useEffect(() => {
+    if (hasToken && !isAuthorized && !isLoading) {
+      fetchUser();
+    }
+  });
```

Comment:
You need to send the second param to this useEffect - when do you need to call it? Now it will be called with each rerender